parquet/src/main/java/org/apache/iceberg/parquet/ParquetWriter.java

```diff
@@ -46,7 +46,7 @@
 
 class ParquetWriter<T> implements FileAppender<T>, Closeable {
 
-  private static DynConstructors.Ctor<PageWriteStore> pageStoreCtorParquet = DynConstructors
+  private static final DynConstructors.Ctor<PageWriteStore> pageStoreCtorParquet = DynConstructors
```

Comment:
In this PR, I also fixed this compiler warnings and `RuntimeIOException` deprecation (to `UncheckedIOException`). Please let me know if it is preferred to leave those out as a separate PR

Reply:
Thanks for working on that! I think we are accumulating more and more of these warnings, better to fix them as much as possible along the way.